### DISCUSSION TITLE
FreeRTOS for ESP32

### DISCUSSION THREAD

**Discussion by Rita858** [feilipu/Arduino_FreeRTOS_Library#79] [Q&A]
Hi,
You mentioned that this lib implemented for AVR(UNO, Leonardo, Mega). 
But I found that it can be used on ESP32 successfully.
I am so worried about whether it would have a hidden mistake here?

**Comment by feilipu** [comment]
Yes there are [assembly language routines](https://github.com/feilipu/Arduino_FreeRTOS_Library/blob/master/src/portable.h) that only work with AVR ATmega in this port. So something will be not working.

Note that FreeRTOS is part of the SDK for ESP32, so perhaps your code is using SDK functions instead?
